sdk/src/keyed_account.rs

```diff
@@ -186,11 +186,14 @@ where
         .collect::<A>()
 }
 
-/// Return the next KeyedAccount or a NotEnoughAccountKeys error
-pub fn next_keyed_account<'a, 'b, I: Iterator<Item = &'a KeyedAccount<'b>>>(
```

Comment:
@garious I could go either way on the iter vs index debate.  Both have access functions that provide sufficient protection.  Also, our instruction definitions spcify the accounts numerically, so it kinda lends itself to index-based accessing as well.  We might also want to provide some kind of (maybe generated) accessor functions along with the instruction definitions to index a specific account of an instruction.  Doing this by index also eliminates the need to convert to an iter to grab some accounts, and then badk to a slice to pass off to other instructions.  One place where it isn't as nice is in instructions where there is a group of accounts that will be iterated through but the length may not be known, like list of multisig acounts.

Reply:
I can stop blocking progress here.  My hope was that there was just some lifetime issue that needed to be solved and then the original code could be left alone.  Sounds like the situation is more nuanced.  Feel free to move forward in whatever way you feel makes the most sense.